### ISSUE TITLE
Underlying mechanism of change detection and ngzone

### ISSUE THREAD

**Issue by ganeshkbhat** [angular/angular#17169]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ x] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

Some silly question.

1) What is the exact way change is detected in Angular. I tried to check with conflicting or confusion response mixes. 
[Change detection flow](https://stackoverflow.com/questions/44298214/change-detection-api-underlying-architecture-in-angular)
[Understanding change detection](https://stackoverflow.com/questions/42982784/understanding-change-detection-in-angular-2)

I read some places that
a) Change detector object in every component obj and relays finished asynchronous actions with the onTurnDone event
[Blog Understanding Change detection](https://auth0.com/blog/understanding-angular-2-change-detection/)
[Blog change detection explained](https://blog.thoughtram.io/angular/2016/02/22/angular-2-change-detection-explained.html#who-notifies-angular)

b) OR is it a manual check on viewstate (currView.state |= ViewState.ChecksEnabled;) and this is dirty checked (not a listener)
[Blog Change detection](https://hackernoon.com/everything-you-need-to-know-about-change-detection-in-angular-8006c51d206f)
[Blog Change detection of Angular](https://hackernoon.com/angulars-digest-is-reborn-in-the-newer-version-of-angular-718a961ebd3e)

I do understand that tick() in applicationref is triggered. But how and whats the flow. Is there any official document how ngZone and Change Detection are related and work? I understand this might be a wrong place to ask this but seems like I had no option. Any help is welcome.

**Expected behavior**

n/a
**Minimal reproduction of the problem with instructions**
n/a

**What is the motivation / use case for changing the behavior?**

Understand the right way of change detection

**Please tell us about your environment:**
Windows 10, Node 7.9, Angular 4

* **Angular version:** 4.X

* **Browser:** [all  ]
 
* **Language:** [all | TypeScript 2.1]

* **Node (for AoT issues):** `node --version` =   
v 7.9

**Comment by maximusk** [comment]
You got an [answer on stackoverflow](https://stackoverflow.com/a/44298962/2545680), what is unclear about it?

> I tried to check with conflicting or confusion response mixes.

Where are they conflicting?

**Comment by ganeshkbhat** [comment]
The explanation in [Change Detection document](https://blog.thoughtram.io/angular/2016/02/22/angular-2-change-detection-explained.html#who-notifies-angular) says the following:

`The short version is, that somewhere in Angular’s source code, there’s this thing called ApplicationRef, which listens to NgZones onTurnDone event. Whenever this event is fired, it executes a tick() function which essentially performs change detection.`

But the stackoverflow it mentions there is no event that triggers change and it is dirty checking.
1) `No, change detector is not a listener. Each component in Angular is represented as a view. Hence the application is a tree of views.`

2) `How is it that the change basically capturing the state change? Using listeners? - no, it's dirty checking mechanism. Read my another article :). It basically remembers the previous values and compares them with the current value during the check`

Need to understand now whether it is 

`NgZones onTurnDone event` > `dirty checking` of view change values
or 
`dirty checking` of view change values (if so how is the dirty checking triggered? or initiated ?)

Is there a document identifying the flow of changedetection?

**Comment by maximusk** [comment]
```
Need to understand now whether it is

NgZones onTurnDone event > dirty checking of view change values
```

It is actually `onMicroTaksEmpty` event now, not `onTurnDone`. Angular subscribes to this event and when notified triggers change detection by [ApplicationRef.tick()](https://github.com/angular/angular/blob/master/packages/core/src/application_ref.ts#L550). This method triggers [detectChanges](https://github.com/angular/angular/blob/master/packages/core/src/application_ref.ts#L558) on the root view. Beyond that, applicationRef.tick() and NgZone mentioned here:

> there’s this thing called ApplicationRef, which listens to NgZones onTurnDone event. Whenever this event is fired, it executes a tick() function which essentially performs change detection.

are irrelevant. What matters then is this:

> Each component in Angular is represented as a view. Hence the application is a tree of views. Angular performs change detection for each relevant node in this view by dirty checking.

**Comment by chuckjaz** [comment]
Hello, we reviewed this issue and determined that it doesn't fall into the bug report or feature request category. This issue tracker is not suitable for support requests, please repost your issue on [StackOverflow](http://stackoverflow.com/) using tag `angular`.

If you are wondering why we don't resolve support issues via the issue tracker, please [check out this explanation](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-got-a-question-or-problem).

**Comment by ganeshkbhat** [comment]
Fair enough. However, this was actually a request for documentation need. Hopefully this will be taken forward.

**Comment by chuckjaz** [comment]
The issue reads like a support request. Can you rephrase it as a aio feature request?
